semgrep_output_v1.atd

```diff
@@ -554,52 +554,62 @@ type rule_times
   <ocaml attr="deriving show">
   <python decorator="dataclass(frozen=True)"> = {
   rule_id: rule_id;
+  (* TODO: time to parse what? *)
   parse_time: float;
   match_time: float;
 }
 
 (* ----------------------------- *)
 (* CLI *)
 (* ----------------------------- *)
-(* LATER: could merge with core_timing above
- * coupling: if you change the JSON schema below, you probably need to
+(* coupling: if you change the JSON schema below, you probably need to
  * also modify perf/run-benchmarks. Run locally  $ ./run-benchmarks --dummy --upload
  *)
-type cli_timing <ocaml attr="deriving show"> = {
-    rules: rule_id_dict list;
+type profile <ocaml attr="deriving show"> = {
+    (* List of rules, including the one read but not run on any target.
+     * This list is actually more an array which allows other
+     * fields to reference rule by number instead of rule_id
+     * (e.g., match_times further below) saving space in the JSON.
+     *
+     * Upgrade note: this used to be defined as a rule_id_dict where
+     * each rule_id was inside a {id: rule_id} record (maybe so
+     * we could have given more profiling info about each rule, but
+     * this was never used, so better to make it simple).
```

Comment:
Ah yeah that was because we used to record a parse time for each rule; this is better now. Feel free to delete this note